src/software/new_geom/util/distance.h

```diff
@@ -0,0 +1,22 @@
+#pragma once
+
+#include "software/new_geom/line.h"
+#include "software/new_geom/point.h"
+
+/**
+ * Finds the shortest distance between a Line and a Point
+ *
+ * @param first the Line
+ * @param second the Point
+ * @return the shortest distance between first and second
+ */
+double distance(const Line &first, const Point &second);
+
+/**
+ * Finds the shortest distance between a Point and a Line
+ *
+ * @param first the Point
+ * @param second the Line
+ * @return the shortest distance between first and second
+ */
+double distance(const Point &first, const Line &second);
```

Comment:
@garethellis0 thoughts on if we want to duplicate javadoc comments for functions iike this where we simply flip the parameters? This is quite common with our utility function so I'm fine with them sharing the same comment and just being right next to one another.

eg.
```
/**
 * Finds the shortest distance between a Line and a Point
 *
 * @param first the Line
 * @param second the Point
 * @return the shortest distance between first and second
 */
double distance(const Line &first, const Point &second);
double distance(const Point &first, const Line &second);
```

Reply:
Agreed, this seems like a reasonable idea.

Reply:
(as long as formatting doesn't insert an empty line between them).

Reply:
Done, clang_format doesn't seem to have any problems with it